instrumentation/testing/logger.go

```diff
@@ -0,0 +1,184 @@
+	}
+
+	// We extract all methods pointer of Test, to avoid logging twice
+	var test *Test
+	typeOfTest := reflect.TypeOf(test)
+	for i := 0; i < typeOfTest.NumMethod(); i++ {
+		method := typeOfTest.Method(i)
+		skippedPointers[method.Func.Pointer()] = true
+	}
+}
+
+func PatchTestingLogger() {
+	patchPointersMutex.Lock()
+	defer patchPointersMutex.Unlock()
+	patchError()
+	patchErrorf()
+	patchFatal()
+	patchFatalf()
+	patchLog()
+	patchLogf()
+	patchSkip()
+	patchSkipf()
+}
+
+func UnpatchTestingLogger() {
+	patchPointersMutex.Lock()
+	defer patchPointersMutex.Unlock()
+	for _, patch := range patches {
+		logOnError(patch.Unpatch())
+	}
+	patches = nil
+	patchPointers = map[uintptr]bool{}
+}
+
+func patchError() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		args := getArgs(argsValues[0])
+		test.Error(args...)
+	}
+	patch("Error", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+
+}
+func patchErrorf() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		format := argsValues[0].String()
+		args := getArgs(argsValues[1])
+		test.Errorf(format, args...)
+	}
+	patch("Errorf", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+func patchFatal() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		args := getArgs(argsValues[0])
+		test.Fatal(args...)
+	}
+	patch("Fatal", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+func patchFatalf() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		format := argsValues[0].String()
+		args := getArgs(argsValues[1])
+		test.Fatalf(format, args...)
+	}
+	patch("Fatalf", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+func patchLog() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		args := getArgs(argsValues[0])
+		test.Log(args...)
+	}
+	patch("Log", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+func patchLogf() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		format := argsValues[0].String()
+		args := getArgs(argsValues[1])
+		test.Logf(format, args...)
+	}
+	patch("Logf", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+func patchSkip() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		args := getArgs(argsValues[0])
+		test.Skip(args...)
+	}
+	patch("Skip", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+func patchSkipf() {
+	fn := func(test *Test, argsValues []reflect.Value) {
+		format := argsValues[0].String()
+		args := getArgs(argsValues[1])
+		test.Skipf(format, args...)
+	}
+	patch("Skipf", fn)
+	patchPointers[reflect.ValueOf(fn).Pointer()] = true
+}
+
+func getArgs(in reflect.Value) []interface{} {
+	var args []interface{}
+	if in.Kind() == reflect.Slice {
+		for i := 0; i < in.Len(); i++ {
+			args = append(args, in.Index(i).Interface())
+		}
+	}
+	return args
+}
+
+func patch(methodName string, methodBody func(test *Test, argsValues []reflect.Value)) {
+	if method, ok := commonPtr.MethodByName(methodName); ok {
+		var patch *mpatch.Patch
```

Comment:
```suggestion
		var methodPatch *mpatch.Patch
```

too many `patch`es everywhere